### PR TITLE
Add "checkout_quantity_changed" webhook

### PR DESCRIPTION
**Pull request by derenio**
This `PR` adds a `checkout_quantity_changed` webhook that is used by _Klaviyo extention_ to track what customers add to their baskets.

### Pull Request Checklist

1. [x] The changes are tested.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 46807a54ba8e0190c803fa62df05949f109edc57 (derenio/saleor @ feature/checkout_webhook)
Base comparison is 440c9b18fce96a14827a59ec1c4c659c0fdf1a09.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	         50	         50	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/webhook/__init__.py [hunk 1]
```diff
@@ -68,4 +78,5 @@ class WebhookEventType:
         ORDER_FULFILLED: "order.manage_orders",
         CUSTOMER_CREATED: "account.manage_users",
         PRODUCT_CREATED: "product.manage_products",
+        CHECKOUT_QUANTITY_CHANGED: "order.manage_orders",
```
Comment (korycins) [inline saleor/webhook/__init__.py]: `"order.manage_orders"`?

Comment (maarcingebala) [reply saleor/webhook/__init__.py]: We migrated to enums to represent permissions, so it will be `OrderPermissions.MANAGE_ORDERS`.

Comment (korycins) [reply saleor/webhook/__init__.py]: @maarcingebala Should we use order permissions for the checkout's actions?

Comment (maarcingebala) [reply saleor/webhook/__init__.py]: Normally checkout doesn't use any permissions, so maybe it's not required at all?

Comment (derenio) [reply saleor/webhook/__init__.py]: @maarcingebala I've used the `order` permission as there isn't a one for the `checkout`. This webhook allows an extension to capture all the users' checkouts so it definitely requires some permission as we don't want everyone to be able to get this information.

I've picked `order` as I thought if you already can listen for the `orders` then `checkouts` don't have more information.

Comment (maarcingebala) [reply saleor/webhook/__init__.py]: I wouldn't mix these. Maybe simply need new permission for fetching checkouts?

Comment (derenio) [reply saleor/webhook/__init__.py]: @maarcingebala I've added new `CheckoutPermission` and used it for this checkout's webhook. As for why we need it. We don't want every extension to be able to listen for the `checkout_quantity_changed` webhook as it exposes users  baskets. That should be explicit and possible only for dedicated extensions with proper permissions.

File: saleor/webhook/payload_serializers.py [hunk 2]
```diff
@@ -18,9 +18,13 @@ class PayloadSerializer(JSONSerializer):
     def __init__(self):
         super().__init__()
         self.additional_fields = {}
+        self.extra_dict_data = {}
+        self.obj_id_name = "id"
```
Comment (korycins) [inline saleor/webhook/payload_serializers.py]: Shouldn't be `None` as we assign value in `serialize` method?

File: saleor/webhook/payloads.py [hunk 3]
```diff
@@ -96,6 +98,41 @@ def generate_order_payload(order: "Order"):
     return order_data
 
 
+def generate_checkout_payload(checkout: "Checkout"):
+    serializer = PayloadSerializer()
+    checkout_fields = (
+        "created",
+        "last_change" "status",
```
Comment (korycins) [inline saleor/webhook/payloads.py]: missing comma

File: setup.cfg [hunk 4]
```diff
@@ -12,6 +12,7 @@ exclude_lines =
     pragma: no cover
     raise NotImplementedError
     return NotImplemented
+    if TYPE_CHECKING:
```
Comment (korycins) [inline setup.cfg]: Is it expected?

Comment (derenio) [reply setup.cfg]: This was added because `codecov` treated the code behind `if` as not covered. Now it is ignoring it.

**Pull request merged**